src/Net/SocketExtensions.cs

```diff
@@ -22,9 +22,39 @@ namespace Amqp
     using System.Net;
     using System.Net.Sockets;
     using System.Threading.Tasks;
+    using System.Runtime.InteropServices;
 
     static class SocketExtensions
     {
+        public static bool SetTcpKeepAlive(this Socket socket, ulong keepAliveTime, ulong keepAliveInterval)
+        {
+            /* the native structure
+            struct tcp_keepalive {
+            ULONG onoff;
+            ULONG keepalivetime;
+            ULONG keepaliveinterval;
+            };
+            */
+
+            try
+            {
+                ulong dummy = 0;
+                byte[] inOptionValues = new byte[Marshal.SizeOf(dummy) * 3];
```

Comment:
Marshal is not available on netstandard (.Net Core). We can get the byte[] of each field and merge them into one.